mbed_project/_internal/project_data.py

```diff
@@ -97,3 +97,25 @@ def from_existing(cls, root_path: Path) -> "MbedProgramData":
             raise ValueError(f"An Mbed program does not exist at {root_path}")
 
         return cls(config_file=config, mbed_file=mbed_file)
+
+
+@dataclass
+class MbedOS:
+    """Metadata associated with a copy of MbedOS.
+
+    This object holds information about MbedOS used by MbedProgram.
+
+    Attributes:
+        targets_json_file: Path to a targets.json file, which contains target data specific to MbedOS revision.
+    """
+
+    targets_json_file: Path
+
+    @classmethod
+    def from_existing(cls, root_path: Path) -> "MbedOS":
+        """Create MbedOS from a directory constaining an existing MbedOS installation."""
+        targets_json_file = root_path / "targets.json"
```

Comment:
Should the path not be `root_path / "targets" / "targets.json"`?

Reply:
But of course. 🤦‍♂

Reply:
https://github.com/ARMmbed/mbed-project/pull/5/commits/b6fc20473d1ad00f7d8e6d2d1bc7b730ce2edf34